src/codegen/llvm/codegen_llvm_visitor.cpp

```diff
@@ -406,7 +662,7 @@ void CodegenLLVMVisitor::visit_codegen_function(const ast::CodegenFunction& node
     block->accept(*this);
 
     // If function has a void return type, add a terminator not handled by CodegenReturnVar.
-    if (node.is_void())
+    if (node.get_return_type()->get_type() == ast::AstNodeType::VOID)
```

Comment:
Ah yes! I might have missed this in code review as well - return type is of type `CodegenVarType`.